### PR TITLE
feat: add cucumber step name in Sauce

### PR DESCRIPTION
**Pull request by wswebcreation**
## Proposed changes
Cucumber has a feature, scenario and a step. We now only push the feature and scenario name to Sauce Labs. This PR will also push the test step. This makes it "easier" to debug which command belongs to which step, see also below.

**NOTE:** I've added `--` to have an _indentation_ in Sauce, spaces will be trimmed and `&nbsp;` will not be translated to spaces in Sauce Labs

image

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments
-

### CODE REVIEW

File: packages/wdio-sauce-service/src/service.ts [hunk 1]
```diff
@@ -230,7 +230,7 @@ export default class SauceService implements Services.ServiceInstance {
             return
         }
 
-        return this.setAnnotation(`sauce:context=Feature:${this._suiteTitle}`)
+        return this.setAnnotation(`sauce:context=Feature: ${this._suiteTitle}`)
```
Comment (erwinheitzman) [inline packages/wdio-sauce-service/src/service.ts:233]: is this added space intentional?

**Comment by erwinheitzman** [comment]
How about we add a single `-` for Scenario's so it becomes a logical tree like so:
```
Feature
-Scenario
--Step
```

**Comment by wswebcreation** [comment]
> How about we add a single `-` for Scenario's so it becomes a logical tree like so:
> 
> ```
> Feature
> -Scenario
> --Step
> ```

Looks fine for me, I'll fix that and push it

**Review by erwinheitzman — APPROVED** [review]

**Review by christian-bromann — APPROVED** [review]
Very nice 👍

**Pull request merged**